presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -163,6 +163,12 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
         Table table = metastore.getTable(tableName.getSchemaName(), tableName.getTableName())
                 .orElseThrow(() -> new TableNotFoundException(tableName));
 
+        // verify table is not marked as non-readable
+        String tableNotReadable = table.getParameters().get(OBJECT_NOT_READABLE);
```

Comment:
I see. So the key is to perform these checks before the short circuit when we don't have any partitions (one line 176)

Reply:
Correct. Which is why it took a bit of refactoring, since previously we needed a partition in order to fetch the table metadata.